File: EASSET.Web/Modules/ServerTypes/MasterData/SubKegiatanRow.ts
import { fieldsProxy } from "@serenity-is/corelib";

export interface SubKegiatanRow {
    Id?: number;
    BidangView?: string;
    UrusanView?: string;
    ProgView?: string;
    KegView?: string;
    SubKegView?: string;
    SubKeg50Id?: string;
    SubKeg50Kode?: string;
}

export abstract class SubKegiatanRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'SubKegView';
    static readonly localTextPrefix = 'MasterData.SubKegiatan';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<SubKegiatanRow>();
}

File: EASSET.Web/Modules/MasterData/SubKegiatan/SubKegiatanRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace EASSET.MasterData;

[ConnectionKey("Default"), Module("MasterData"), TableName("vwSubKegiatan")]
[DisplayName("Sub Kegiatan"), InstanceName("Sub Kegiatan"), GenerateFields]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
public sealed partial class SubKegiatanRow : Row<SubKegiatanRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), IdProperty]
    public long? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Bidang View"), Size(261), QuickSearch]
    public string BidangView { get => fields.BidangView[this]; set => fields.BidangView[this] = value; }

    [DisplayName("Urusan View"), Size(121), QuickSearch]
    public string UrusanView { get => fields.UrusanView[this]; set => fields.UrusanView[this] = value; }

    [DisplayName("Prog View"), Size(286), QuickSearch]
    public string ProgView { get => fields.ProgView[this]; set => fields.ProgView[this] = value; }

    [DisplayName("Keg View"), Size(1046), QuickSearch]
    public string KegView { get => fields.KegView[this]; set => fields.KegView[this] = value; }

    [DisplayName("Sub Keg View"), Size(2566), QuickSearch, NameProperty]
    public string SubKegView { get => fields.SubKegView[this]; set => fields.SubKegView[this] = value; }

    [DisplayName("Sub Keg50 Id"), Column("subKeg50ID"), Size(20), NotNull]
    public string SubKeg50Id { get => fields.SubKeg50Id[this]; set => fields.SubKeg50Id[this] = value; }

    [DisplayName("Sub Keg50 Kode"), Size(20)]
    public string SubKeg50Kode { get => fields.SubKeg50Kode[this]; set => fields.SubKeg50Kode[this] = value; }
}